### ISSUE TITLE
Plan name does not change in Admin portal

### ISSUE THREAD

**Issue by alexdrenea** [Azure/Commercial-Marketplace-SaaS-Accelerator#345]
**Describe the bug**
When changing plans (either customer initiated or partner initiated) the plan name is not updated in the admin portal subscriptions page or details.
This can be confusing and worrying for the partner as they can't tell if their change has taken effect or not.

**To Reproduce**
Change plan for one of the subscriptions 

**Expected behavior**
Once change plan is complete, I would like to see the new plan name in the offers page.

**Screenshots**
![image](https://user-images.githubusercontent.com/8275679/199807777-e7459cf9-de3e-4e06-a155-dc6bb7f99b10.png)

**Additional Context**
Ideally the change would happen automatically once the plan is changed, however, it would be acceptable if this would work when clicking "Fetch all Subscriptions" button

**Comment by alexdrenea** [comment]
@santhoshb-msft - I can work on this if you don't mind. I have a branch that I started for this. no need to duplicate efforts

**Comment by santhoshb-msft** [comment]
Sounds good @alexdrenea !

**Comment by santhoshb-msft** [comment]
But there is the webhook flow I would like to mention, lets connect offline.

**Comment by santhoshb-msft** [comment]
@alexdrenea,  can you help with two things please
1. On fetch lets sync all three when different, like state/plan/quantity
2. and add an audit log as well
